src/plugins/grpc/server.js

```diff
@@ -0,0 +1,137 @@
+'use strict'
+
+const { FORMAT_HTTP_HEADERS, Tags } = require('opentracing')
+
+let patchedHandlers = []
+
+function createWrapHandler (grpc, tracer, config, handler) {
+  return function wrapHandler (func) {
+    return function funcHandler (emitter) {
+      const { metadata, request } = emitter
+      const { type } = this
+      const isStream = ['client_stream', 'server_stream'].includes(type)
+      const childOf = tracer.extract(FORMAT_HTTP_HEADERS, metadata.getMap())
+      const span = tracer.startSpan('grpc.server', {
+        childOf,
+        tags: {
+          [Tags.SPAN_KIND]: Tags.SPAN_KIND_RPC_SERVER,
+          'grpc.method_name': handler,
+          'grpc.method_type': type,
+          'resource.name': handler.split('/').slice(-1),
+          'service.name': config.service || `${tracer._service}-grpc`
+        }
+      })
+
+      tracer.scopeManager().activate(span)
+
+      if (request) {
+        if (metadata) {
+          for (const [key, value] of Object.entries(metadata.getMap())) {
+            span.setTag(`grpc.metadata.${key}`, value)
+          }
+        }
+
+        if (config.arguments) {
+          const values = config.arguments(request)
+
+          for (const key in values) {
+            span.setTag(`grpc.arguments.${key}`, values[key])
+          }
+        } else {
+          span.setTag('grpc.arguments', request)
+        }
+      }
+
+      if (isStream) {
+        emitter.on('error', err => {
+          handleError(config, span, err)
+
+          span.finish()
+        })
+
+        // Finish the span if the response only if it was successful.
+        // Otherwise it'll be finished in the `error` listener.
+        emitter.on('finish', function () {
+          if (emitter.status.code === 0) {
+            span.finish()
+          }
+        })
+
+        // Call the original stream request, without modification.
+        return func.apply(this, arguments)
+      }
+
+      const [, callback] = arguments
+
+      // Call the unary request with a wrapped callback.
+      return func.call(this, emitter, (err, value, trailer, flags) => {
+        if (err) {
+          handleError(config, span, err, trailer)
+        }
+
+        span.finish()
+
+        if (callback) {
+          callback.call(this, value, trailer, flags)
+        }
+      })
+    }
+  }
+}
+
+function handleError (config, span, err, metadata) {
+  span.addTags({
+    'error.code': err.code,
```

Comment:
Any way to also get the type? If not, hardcode `error.type` to `Error`.

Reply:
Added 👍